plugins/git-commit/git-commit.plugin.zsh

```diff
@@ -14,6 +14,56 @@ _git_commit_aliases=(
   'wip'
 )
 
+_git_commit_flags=(
+  '-s'
+  '--scope'
+  '-a'
+  '--attention'
+  '-sa'
+  '-as'
+)
+
+remove_flags() {
+  local regex=$(IFS="|"; echo "${_git_commit_flags[*]}")
+
+  echo $(sed -E "s/\s($regex)\s|\s($regex)$//g" <<< "$1")
```

Comment:
It should be enough to use `*` (matches between zero and unlimited times!). Let's add `-n` as well!
```suggestion
  echo -n $(sed -E "s/\s*($regex)\s*//g" <<< "$1")
```